crmprtd/align.py

```diff
@@ -8,12 +8,222 @@
+""" # noqa
+    q = sesh.execute(query_txt, {
+        'x': lon,
+        'y': lat,
+        'thresh': threshold}
+    )
+    valid_hid = set([x[0] for x in q.fetchall()])
+    log.debug("history_ids in threshold", extra={'hid': valid_hid})
+    return valid_hid
+
+
+def convert_unit(val, src_unit, dst_unit):
+    log.debug('Converting units', extra={'src_unit': src_unit,
+                                         'dst_unit': dst_unit})
+    try:
+        val = Q_(val, ureg.parse_expression(src_unit))  # src
+        val = val.to(dst_unit).magnitude  # dest
+    except Exception as e:
+        log.error('Unable to convert units', extra={'src_unit': src_unit,
+                                                    'dst_unit': dst_unit,
+                                                    'exception': e})
+        raise e
+    log.debug('Converted units')
+    return val
+
+
+def unit_db_check(unit_obs, unit_db, val):
+    log.debug('Check if units match')
+    if unit_obs != unit_db:
+        try:
+            val_conv = convert_unit(val, unit_obs, unit_db)
+        except Exception as e:
+            log.error('Unable to convert units',
+                      extra={'src_unit': unit_obs,
+                             'dst_unit': unit_db,
+                             'exception': e})
+            return None
+        return val_conv
+    else:
+        return val
+
+
+def unit_check(sesh, unit_obs, unit_db, val):
+    log.debug('Check if there are units to compare')
+    if not unit_obs and not unit_db:
+        return None
+    else:
+        return unit_db_check(unit_obs, unit_db, val)
+
+
+def match_station_with_active(sesh, obs_tuple, histories):
+    log.debug('Search for active stations')
+    for history in histories:
+        hist = sesh.query(History).filter(
+            and_(History.id == history.id,
+                 History.sdate != None,
+                 History.edate == None))    # noqa
+
+        if hist.count() == 1:
+            log.debug('Matched history',
+                      extra={'station_name': hist.first().station_name})
+            return hist.first()
+
+    log.warning('Could not find match',
+                extra={'network_name': obs_tuple.network_name,
+                       'native_id': obs_tuple.station_id})
+    return None
+
+
+def match_station_with_location(sesh, obs_tuple, histories):
+    log.debug('Find matching station with location')
+    close_stns = closest_stns_within_threshold(sesh, obs_tuple.lon,
+                                               obs_tuple.lat, 800)
+
+    if len(close_stns) == 0:
+        log.debug('No station nearby')
+        return create_station_and_history_entry(sesh, obs_tuple)
+
+    for id in close_stns:
+        for history in histories:
+            if id == history.id:
+                log.debug('Matched history',
+                          extra={'station_name': history.station_name})
+                return history
+
+
+def match_station(sesh, obs_tuple, history):
```

Comment:
Since this is only called from the `>=2` branch of the conditional, we should assume that it is being called with multiple history objects. So rename `history` to `histories` and keep the number consistent across the code base.